test/test_auth.py

```diff
@@ -576,3 +578,219 @@ async def test_change_server_password_server_lifetime(bouncer, pg):
+    # CN expected in map is "bouncer" which matches the CN in the client cert
+    # hence the test succeeds.
+    bouncer.psql_test(
+        dbname="p0",
+        host="localhost",
+        user="bouncer",
+        sslmode="verify-full",
+        sslkey=client_key,
+        sslcert=client_cert,
+        sslrootcert=root,
+    )
+
+    # The client connects to p0y using a client certificate with CN=bouncer.
+    # hba_eval returns the followign line:
+    #    hostssl p0y             all             0.0.0.0/0               cert    map=test2
+    # where
+    #   test2           bouncer                 all
+    #   test2           pgbouncer.acme.org      "anotheruser"
+    # test2 mapping allows any client with CN="bouncer" to connect using any user name.
+    # Hence the test succeeds.
+    bouncer.psql_test(
+        dbname="p0y",
+        host="localhost",
+        user="someuser",
+        sslmode="verify-full",
+        sslkey=client_key,
+        sslcert=client_cert,
+        sslrootcert=root,
+    )
+
+    client_key = cert_dir / "TestCA1" / "sites" / "04-pgbouncer.acme.org.key"
+    client_cert = cert_dir / "TestCA1" / "sites" / "04-pgbouncer.acme.org.crt"
+
+    # The client connects to p0y using a client certificate with CN=pgbouncer.acme.org.
+    # hba_eval returns the followign line:
+    #    hostssl p0y             all             0.0.0.0/0               cert    map=test2
+    # where
+    #   test2           bouncer                 all
+    #   test2           pgbouncer.acme.org      "anotheruser"
+    # for CN=pgbouncer.acme.org, test2 allows to use anotheruser. Hence the test fails.
+
+    with pytest.raises(
+        subprocess.CalledProcessError,
+    ):
+        with bouncer.log_contains(
+            "p0y/someuser@127.0.0.1:39712 ident map: test2 does not have a match"
+        ):
+            bouncer.psql_test(
+                dbname="p0y",
+                host="localhost",
+                user="someuser",
+                sslmode="verify-full",
+                sslkey=client_key,
+                sslcert=client_cert,
+                sslrootcert=root,
+            )
+
+    # The client connects to p0y using a client certificate with CN=pgbouncer.acme.org.
+    # hba_eval returns the followign line:
+    #    hostssl p0y             all             0.0.0.0/0               cert    map=test2
+    # where
+    #   test2           bouncer                 all
+    #   test2           pgbouncer.acme.org      "anotheruser"
+    # for CN=pgbouncer.acme.org, test2 allows to use anotheruser. Hence the test succeeds.
+
+    bouncer.psql_test(
+        dbname="p0y",
+        host="localhost",
+        user="anotheruser",
+        sslmode="verify-full",
+        sslkey=client_key,
+        sslcert=client_cert,
+        sslrootcert=root,
+    )
+
+
+@pytest.mark.skipif("WINDOWS", reason="Windows does not have SIGHUP")
+def test_peer_auth_ident_map(bouncer):
+    cur_user = getpass.getuser()
+
+    with open("ident.conf", "w") as f:
```

Comment:
These files should be created in the config directory of pgbouncer. Otherwise they polute the current working directory when running the tests locally.